src/featureFlag/index.tsx

```diff
@@ -0,0 +1,23 @@
+import { createContext, ReactNode, useContext } from 'react'
+
+interface FeatureFlagContextType {
+  isLoaded: boolean
+}
+
+const DEFAULT_STATE = { isLoaded: false }
+const FeatureFlagContext = createContext<FeatureFlagContextType>(DEFAULT_STATE)
+
+export function FeatureFlagProvider({ children }: { children: ReactNode }) {
+  return <FeatureFlagContext.Provider value={{ isLoaded: false }}>{children}</FeatureFlagContext.Provider>
```

Comment:
this provider will also own not rendering the app until the flags are fetched (or a timeout has been hit)